src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs

```diff
@@ -8746,5 +8746,1770 @@ public async Task SelectPhoneMethodTypeDuringChallenge()
+                ""type"": ""phone"",
+                ""key"": ""phone_number"",
+                ""id"": ""auttzfsi4eiZIdLK85d6"",
+                ""displayName"": ""Phone"",
+                ""methods"": [
+                    {
+                        ""type"": ""sms""
+                    }
+                ]
+            }
+        ]
+    },
+    ""authenticatorEnrollments"": {
+        ""type"": ""array"",
+        ""value"": [
+            {
+                ""type"": ""password"",
+                ""key"": ""okta_password"",
+                ""id"": ""lae1y7cy0SnqvVzjy5d6"",
+                ""displayName"": ""Password"",
+                ""methods"": [
+                    {
+                        ""type"": ""password""
+                    }
+                ]
+            }
+        ]
+    },
+    ""user"": {
+        ""type"": ""object"",
+        ""value"": {
+            ""id"": ""00uuyydydG04jXoEA5d6""
+        }
+    },
+    ""cancel"": {
+        ""rel"": [
+            ""create-form""
+        ],
+        ""name"": ""cancel"",
+        ""href"": ""https://fake.example.com/idp/idx/cancel"",
+        ""method"": ""POST"",
+        ""produces"": ""application/ion+json; okta-version=1.0.0"",
+        ""value"": [
+            {
+                ""name"": ""stateHandle"",
+                ""required"": true,
+                ""value"": ""02u302Css93ifIFDrXNJQaCJjRSHLmCVaJ41L2jMOZ"",
+                ""visible"": false,
+                ""mutable"": false
+            }
+        ],
+        ""accepts"": ""application/json; okta-version=1.0.0""
+    },
+    ""app"": {
+        ""type"": ""object"",
+        ""value"": {
+            ""name"": ""oidc_client"",
+            ""label"": ""Dotnet IDX Web App"",
+            ""id"": ""xxxxxxxx""
+        }
+    }
+}";
+            #endregion;
+            IIdxContext idxContext = new IdxContext("test code verifier", "test code challenge", "test code challenge method", "test interaction handle", "test state");
+            MockHttpMessageHandler mockHttpMessageHandler = new MockHttpMessageHandler();
+            mockHttpMessageHandler.AddTestResponse("/idp/idx/introspect", introspectResponse);
+            mockHttpMessageHandler.AddTestResponse("/idp/idx/credential/enroll", enrollResponse, System.Net.HttpStatusCode.BadRequest);
+
+            HttpClient httpClient = new HttpClient(mockHttpMessageHandler);
+
+            IdxClient idxClient = new IdxClient(TesteableIdxClient.DefaultFakeConfiguration, httpClient, NullLogger.Instance);
+            EnrollPhoneAuthenticatorOptions enrollPhoneAuthenticatorOptions = new EnrollPhoneAuthenticatorOptions
+            {
+                AuthenticatorId = "testAuthenticatorId",
+                PhoneNumber = "12345678901",
+                MethodType = AuthenticatorMethodType.Sms
+            };
+
+            await Assert.ThrowsAsync<OktaApiException>(() => idxClient.EnrollAuthenticatorAsync(enrollPhoneAuthenticatorOptions, idxContext));
```

Comment:
Does this test pass? The exception should be `OktaException` and not `OktaApiException`.

Reply:
Let's assert the exception message too. You can do something like this:
```
            await function.Should()
                .ThrowAsync<OktaException>()
                .WithMessage("*<ExceptionMessage>*");
```

Reply:
These exceptions are thrown by the abstractions library, I think it would be dangerous to assert the message here because that library is somewhat opaque and the message may change depending on the version of the library.